### ISSUE TITLE
Enhancements to Makefile/build process

### ISSUE THREAD

**Issue by bartoszmajsak** [opendatahub-io/opendatahub-operator#696]
Just a few ideas on how to improve the build process based on my dev experience so far.

### Refine Default Target 

Our default Makefile target should comprehensively handle the workflow which includes tools installation, code generation, manifest generation, linting, testing, and building. That would follow the principle of least surprise, where developers expect the default action to be a complete build cycle.

Simply put - after freshly cloning the project, executing `make` should succeed without any errors.

### Addition of 'Clean' Target: 

Cleans up binaries and other artifacts so that we can regenerate them.

I believe implementing these improvements will not only streamline our development process but also mitigate potential integration and build issues in the future. And above all would make PR reviews more focused on the actual feature being shipped.

**Comment by ykaliuta** [comment]
Is the proposal to make `make image` the default target? Or `image build`?

**Comment by bartoszmajsak** [comment]
> Is the proposal to make `make image` the default target? Or `image build`?

@ykaliuta My line of thinking is. When I clone the repo I execute `make` and I expect to get a working binary. But it's also the target I normally execute as my part of my feedback loop. In my opinion involing the default `make` target should result in correct code and valid binary. So I would image the chain of targets to:

1. set up the whole toolchain I need for daily work 
2. run linters
3. run (fast) tests
4. build the binary (including code generation)

At this point, we are missing `2.` and `3.` and partially `1.`.

WDYT?

**Comment by bartoszmajsak** [comment]
> Got it. Well, I do not think setting up toolchain should be part of it. Others sound fine for me if it's ok for the team

Fair enough, perhaps the targets called should take care of it on their own (e.g. lint or generate will ensure they have needed binaries).

**Comment by ykaliuta** [comment]
> > Got it. Well, I do not think setting up toolchain should be part of it. Others sound fine for me if it's ok for the team
> 
> What I meant by toolchain here is not necessary docker or go but the internal project tools such as golangci-lint, controller-gen, ginkgo (if used) and so on.

Ah, ok. That makes sense, it's already done for some tools.

**Comment by ykaliuta** [comment]
Is this what you want? I have concerns about golangci-lint running with --fix for the scenario.
